src/api/api.ts

```diff
@@ -177,23 +193,74 @@ class API {
     }
   }
 
+  async getClientDetails(acceptRole: string): Promise<AccountDataDTO> {
+    try {
+      const response = await this.authorizedFetch<AccountDataDTO>(
+        "GET",
+        "api/client",
+        { "Accept-Role": acceptRole }
+      );
+      return response;
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        throw new Error(`Failed to fetch caretaker profile: ${error.message}`);
+      }
+      throw new Error(
+        "An unknown error occurred while fetching caretaker profile"
+      );
+    }
+  }
+
+  //TODO: popraw
+  async getCaretakerRatings(
+    email: string,
+    page: number | null,
+    size: number | null,
+    sortDirection: string | null,
+    sortBy: string[] | null
+  ): Promise<CaretakerRatingsResponse> {
+    try {
+      let endpoint = `api/caretaker/${email}/rating`;
```

Comment:
Ten endpoint chyba nie istnieje, powinien być `api/rating/${email}` zamiast `api/caretaker/${email}/rating`.
```suggestion
      let endpoint = `api/rating/${email}` ;
```

Reply:
Faktycznie, to chyba po prostu endpoint się zmienił